assignment/assignment.hs

```diff
@@ -53,7 +57,17 @@ count element list = 0 -- replace me
 -- need to traverse the list once)
 
 evensEqualOdds :: [ Integer ] -> Bool
-evensEqualOdds numbers = False -- replace me
+evensEqualOdds numbers = sumlist numbers==((countodds numbers)*2) -- replace me -- divMod may come in handy
+
+countodds [] = 0
+countodds (x:xs) = fromEnum (remainder)*x+ countodds xs
+   where (result, remainder) = divMod x 2
+sumlist [] = 0
+sumlist (x:xs) = x+sumlist xs
+
+
+                          --unable to tell if this works because Enum is failing me, 
+                          -- also, I don't think this counts as one go so that's on me
 
```

Comment:
That is a really interesting idea: I had to think for a minute to convince myself that sum(list) == 2*sum(odds) means that sum(evens) == sum(odds). It's a nice approach. 

This isn't a place where `fromEnum` helps much. In my solution, I simply traversed the list two-elements at a time:

~~~ hs
evensEqualOdds list =
    evens == odds 
    where
        ( evens, odds ) = getSums list
        getSums :: [ Integer ] -> ( Integer, Integer)
        getSums [] = ( 0, 0)
        getSums (x1 : x2 : xs) =
            ( t1 + x1, t2 + x2)
            where 
                ( t1, t2 ) = getSums xs
~~~

or

~~~ hs
evensEqualOdds list =
  differenceOfPairs list == 0
  where 
    differenceOfPairs [] =  0
    differenceOfPairs  [x] = x
    differenceOfPairs (h1:h2:hs) = h1 - h2 + differenceOfPairs hs